.buildkite/scripts/test.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+go install github.com/elastic/go-licenser@latest
+go get -d -t ./...
+go mod download
+go mod verify
+
+if go mod tidy ; then
+    if [ -z "$(git status --porcelain go.mod go.sum)" ] ; then
+        echo "Go module manifest has not changed."
+    else
+        echo "Go module manifest changed. Run 'go mod tidy'" 1>&2
+        exit 1
+    fi
+fi
+go-licenser -d
+
+if find . -name '*.go' | grep -v vendor | xargs gofmt -s -l | read ; then
+    echo "Code differs from gofmt's style. Run 'gofmt -s -w .'" 1>&2
+    exit 1
+fi
+
+# Run the tests
+go install github.com/jstemmer/go-junit-report@latest
+IS_TEST_FAIL=false
+run_test_prepare_junit() {
+    local temporary_file="build/test-report.out"
+    local junit_output=${1:-test-report.out}
+    local root=${2:-false}
+    local go_env=${3:-''} # e.g. GOARCH=386
+    set +e
+    list="$(go list ./... | grep -v /vendor/)"
+    list_string="${list//$'\n'/ }"
+    if [[ $root == "true" ]]; then
```

Comment:
Where is it needed `root` ? In `.ci/test.sh`, it looks like it is not needed `root`.

Reply:
These tests are indeed running by non-root user in Jenkins, and the current builkite agent has user root inside the container, as a result, the different groups of tests will be running in these two cases. For making it the same, I changed a little bit the script (partially related to all comments)